keps/sig-network/2594-multiple-cluster-cidrs/README.md

```diff
@@ -0,0 +1,980 @@
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Share Resources with Service API](#share-resources-with-service-api)
+    - [Pros](#pros)
+    - [Cons](#cons)
+  - [Nodes Register CIDR Request](#nodes-register-cidr-request)
+    - [Pros](#pros-1)
+    - [Cons](#cons-1)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone /
+release*.
+
+-   [ ] (R) Enhancement issue in release milestone, which links to KEP dir in
+    [kubernetes/enhancements](not the initial KEP PR)
+-   [ ] (R) KEP approvers have approved the KEP status as `implementable`
+-   [ ] (R) Design details are appropriately documented
+-   [ ] (R) Test plan is in place, giving consideration to SIG Architecture and
+    SIG Testing input (including test refactors)
+-   [ ] (R) Graduation criteria is in place
+-   [ ] (R) Production readiness review completed
+-   [ ] (R) Production readiness review approved
+-   [ ] "Implementation History" section is up-to-date for milestone
+-   [ ] User-facing documentation has been created in [kubernetes/website], for
+    publication to [kubernetes.io]
+-   [ ] Supporting documentation—e.g., additional design documents, links to
+    mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Today, when Kubernetes' NodeIPAM controller allocates IP ranges for podCIDRs for
+nodes, it uses a single range allocated to the cluster (cluster CIDR). Each node
+gets a range of a fixed size from the overall cluster CIDR. The size is
+specified during cluster startup time and cannot be modified later on.
+
+Kubernetes' IPAM capabilities are an optional behavior that comes with
+Kubernetes out of the box. It is not required for Kubernetes to function, and
+users may use alternate mechanisms.
+
+This proposal enhances how pod CIDRs are allocated for nodes by adding a new
+CIDR allocator that can be controlled by a new resource `ClusterCIDRRange`. This
```

Comment:
At some point I want to discuss the name - it doesn't roll of the tongue :)

Reply:
How about just CIDRRange?